### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by coker2019**
Different transactions are used.
 - Data is modified and deleted using SQL.
 - Different queries are used to respond to the questions.
 - All the screenshots of transactions and queries are attached below:
species1
species2
species3
savepoint
queries

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -29,3 +29,86 @@ WHERE name <> 'Gabumon';
...
+
+-------------------------------------DELETE and ROLLBACK-------------------------------------------
+
+BEGIN; 
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals
```
Comment (topeogunleye) [inline queries.sql]: Hello,

Great job on your SQL submission! 🌟 It's evident that you've understood the core concept of transactions and how to roll them back, which is crucial in maintaining database integrity. Your code is clear, and the logic is sound. Well done!

However, there are some improvements needed to fully align with the project requirements.

### Suggestions 🧐

1. **Verification of Deletion:**
   - After executing the `DELETE` command, insert a `SELECT` statement to validate whether all records from the `animals` table have indeed been deleted.
     ```sql
     SELECT * FROM animals; -- This should return no rows, verifying the deletion.
     ```

### Corrected Sequence 🛠️

Here’s how the enhanced sequence should look like:

```sql
BEGIN; 

DELETE FROM animals;

-- Verification of Deletion
SELECT * FROM animals; -- Should return no rows, signifying successful deletion.

ROLLBACK;

-- Verification of Rollback
SELECT * FROM animals; -- Should return all initial rows, proving successful rollback.
```

### Conclusion 🌟

You've done exceptionally well, and with a few tweaks, your submission will be spot on! Implementing these suggestions ensures not only adherence to project requirements but also solidifies your understanding of SQL transactions and their implications. Keep up the excellent work, and remember, the journey of coding is filled with continuous learning and improvement! 👏

File: queries.sql [hunk 2]
```diff
@@ -29,3 +29,86 @@ WHERE name <> 'Gabumon';
...
+
+SELECT COUNT(*) AS animal_count FROM animals;
+
+SELECT COUNT(*) AS no_escape_count FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+SELECT name, escape_attempts
+FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```
Comment (topeogunleye) [inline queries.sql]: Hello,

You've made a great attempt with your SQL submission! 🌟 It’s clear you have been paying attention to the syntax and structure of SQL queries. However, there is a mismatch between the provided solution and the project requirements.

Your Code:
```sql
SELECT name, escape_attempts
FROM animals
WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```

This code selects the `name` and `escape_attempts` for the animal(s) with the maximum number of escape attempts, which is not aligned with the requirements.

### Project Requirements 📘
You are required to find out who escapes the most, neutered, or not neutered animals. This involves grouping by the `neutered` column and either using `AVG(escape_attempts)` to find the average escape attempts or `MAX(escape_attempts)` to find the maximum escape attempts for each group (neutered and not neutered).

### Suggested Code 🛠️
Here are the correct SQL statements based on the project requirements:

Option 1 (Using `AVG`):
```sql
SELECT neutered, AVG(escape_attempts) FROM animals
GROUP BY neutered;
```

Option 2 (Using `MAX`):
```sql
SELECT neutered, MAX(escape_attempts) FROM animals
GROUP BY neutered;
```

### Conclusion 🌟

It seems like there was a misunderstanding of the project requirements. Please revise your query to reflect the needed grouping by `neutered` status and the use of either `AVG` or `MAX` functions to fulfill the project’s objective correctly. You’re making good progress, and with a little adjustment, you’ll meet the requirements perfectly. Keep up the hard work and happy coding! 👏

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @coker2019, 

You’ve done a fantastic job on the project! 🎉 Your dedication and hard work are clearly reflected in your submission. There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 

#### 1. **SQL Commands:**
   Your SQL commands are correctly placed in the respective files, ensuring an organized and efficient structure.

#### 2. **Schema.sql:**
   - The addition of a new column named `species` with the string type is accurately done. 👏

   ```sql
   ALTER TABLE animals ADD species varchar(255);
   ```

#### 3. **Data.sql:**
   - Excellent job adding the specified data to the `animals` table, maintaining the correct syntax and structure.
   - All the instructed data was properly inserted into the `animals` table.

#### 4. **Queries.sql:**
   - **Transactions are Well-Structured and Correctly Implemented:**
     Your usage of transactions is meticulous, ensuring changes are made and rolled back accurately.
   - **Comprehensive Transaction Results:**
     All steps in each transaction are correctly done, and you have verified changes efficiently.
   - **Efficient Queries:**
     You’ve developed efficient and correct queries to answer the provided questions.
   - **Screenshots Included:**
     You have adhered to the instructions by including screenshots of your transaction results, which are correct, in the pull request description.

#### 5. **Analysis Queries:**
   - All the queries that help to analyze the data are present and correct, providing accurate insights.
   - Screenshots of the queries' results are included, which verifies the successful execution of the queries.

#### 6. **Maintained Integrity:**
   You’ve maintained the integrity of each SQL file by ensuring that each contains only what’s meant to be in them: 
   - **schema.sql** for changes in table structure.
   - **data.sql** for populating tables with data.
   - **queries.sql** for data analysis queries.

### Conclusion 🎈
This is an excellent submission! 🌟 Your attention to detail, organization of SQL files, and adherence to project requirements are commendable. Keep up the great work and continue to strive for perfection in your coding journey! 🚀 Looking forward to seeing more of your excellent work!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- For future pull requests, please notice that your branch name `querry_update` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-querry_update`. This makes it easier to track the changes and the purpose of each branch.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (coker2019) [reply queries.sql]: done.

File: queries.sql (hunk 2 above)
Comment (coker2019) [reply queries.sql]: neutred1

Comment (coker2019) [reply queries.sql]: done

File: queries.sql (hunk 1 above)
Comment (coker2019) [reply queries.sql]: delete22

File: queries.sql (hunk 2 above)
Comment (topeogunleye) [reply queries.sql]: Thank you for implementing the suggested changes swiftly! 🌟 I’ve reviewed your corrected submission, and it now aligns perfectly with the project requirements. Well done! 👏

File: queries.sql (hunk 1 above)
Comment (topeogunleye) [reply queries.sql]: @coker2019, Thank you for implementing the suggested changes swiftly! 🌟 I’ve reviewed your corrected submission, and it now aligns perfectly with the project requirements. Well done! 👏

File: queries.sql (hunk 2 above)
Comment (coker2019) [reply queries.sql]: Hello @topeogunleye 
Still waiting for reveiw and approval. Dunno if you could help out.
Thanks

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi  @coker2019 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Good Job :+1:
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] N/A

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**